src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -951,12 +1021,17 @@ bool ompl::geometric::SPARSdb::addStateToRoadmap(const base::PlannerTerminationC
 
     /* The whole neighborhood set which has been most recently computed */
     std::vector<Vertex> graphNeighborhood;
+    std::vector<Vertex> gnbhd;
+
     /* The visible neighborhood set which has been most recently computed */
     std::vector<Vertex> visibleNeighborhood;
-
+    std::vector<Vertex> vnbhd;
     ++iterations_;
 
+    //@TODO - Ramy: Test if creating another nbhd to seperate recall from connectivity has positive effects.
     findGraphNeighbors(qNew, graphNeighborhood, visibleNeighborhood);
+    if (denseRoadmap_)
+        findGraphNeighbors(qNew, gnbhd, vnbhd, granularity_);
```

Comment:
are `graphNeighborhood` and `gnbhd` both going to be used, or is it worth only performing one search that's conditional on the `denseRoadmap_` flag (also it should be more obvious that this is a flag)

Reply:
Yes! Both will be used.
gnbhd and vnbhd control the "granularity" of the roadmap, meaning the minimum distance between nodes in the roadmap. Hence they have a very small radius (0.001), where as graphNeighborhood is the neighborhood within which we look to make connections between nodes and also the neighborhood used by the recall planner to find "close" paths in the database to the one being planned.
The algorithm works fine with graphNeighborhood having these two uses, but of course one can investigate more the effect of decoupling these two uses as well (making a third radius/neighborhood).